tests/playwright-test/global-setup.spec.ts

```diff
@@ -410,19 +410,47 @@ test('globalSetup should support multiple', async ({ runInlineTest }) => {
   }, { reporter: 'line' });
   expect(result.passed).toBe(2);
 
-  // behaviour: setups in order, teardowns in reverse order.
-  // setup-returned functions inherit their position, and take precedence over `globalTeardown` scripts.
+  // first setups, then setup callbacks in reverse order.
+  // then teardowns in declared order.
   expect(result.outputLines).toEqual([
     'globalSetup1',
     'globalSetup2',
     'globalSetup3',
     'globalSetup4',
     'test a',
     'test b',
-    'globalSetup3Function',
+    'callback3',
+    'callback1',
+    'globalTeardown1',
     'globalTeardown2',
-    'globalSetup1Function',
-    // 'globalTeardown1' is missing, because globalSetup1Function errored out.
+  ]);
+  expect(result.output).toContain('Error: kaboom');
+});
+
+test('globalTeardown runs even if callback failed', async ({ runInlineTest }) => {
```

Comment:
I think this is already covered by `globalSetup1` from the previous test, isn't it?

Reply:
it is, yes. and it's also covered from the other test I had to touch. but this is the main breaking behaviour change, and I wanted to have an explicit test to cover this.